Source/WebCore/html/HTMLElement.cpp

```diff
@@ -877,14 +878,31 @@ auto HTMLElement::computeDirectionalityFromText() const -> TextDirectionWithStro
         }
     }
 
+    if (RefPtr slot = dynamicDowncast<HTMLSlotElement>(this)) {
+        if (auto* assignedNodes = slot->assignedNodes()) {
+            for (auto& childNode : *assignedNodes) {
+                if (childNode->isTextNode()) {
+                    if (auto direction = childNode->textContent(true).defaultWritingDirection())
+                        return { *direction == U_LEFT_TO_RIGHT ? TextDirection::LTR : TextDirection::RTL, RefPtr { childNode.get() } };
+                } else if (RefPtr childElement = dynamicDowncast<HTMLElement>(childNode.get())) {
+                    auto childDirectionality = childElement->computeDirectionalityFromText();
+                    if (childDirectionality.strongDirectionalityNode)
+                        return childDirectionality;
+                }
+            }
```

Comment:
Don't we need to add a `return` statement here?

Reply:
Good catch, thanks!